Remove processing user gesture check before displaying prompt for usdz

JonWBedard opened the pull request:
#### b35f3f7eba94a42a8108ea651f1de8ea4a906402
<pre>
Remove processing user gesture check before displaying prompt for usdz
<a href="https://bugs.webkit.org/show_bug.cgi?id=272321">https://bugs.webkit.org/show_bug.cgi?id=272321</a>
radar://126078233

Reviewed by Alexey Proskuryakov.

Opening a usdz blob isn&apos;t considered to be processing a user gesture,
so having the UserGestureIndicator::processingUserGesture() check results
in the download path being taken instead of the prompt and open to ARQL.

Since we already have the prompt, we can remove the user gesture check.

* Source/WebCore/html/HTMLAnchorElement.cpp:
(WebCore::HTMLAnchorElement::handleClick):

Originally-landed-as: 272448.918@safari-7618-branch (fe35964ae2b9). <a href="https://rdar.apple.com/128280210">rdar://128280210</a>
Canonical link: <a href="https://commits.webkit.org/279000@main">https://commits.webkit.org/279000@main</a>
</pre>
----------------------------------------------------------------------
#### f1cfd5a3139a0e18dcf0279802c0867174696beb
<pre>
Potential UAF in SystemPreviewController::begin
<a href="https://bugs.webkit.org/show_bug.cgi?id=272342">https://bugs.webkit.org/show_bug.cgi?id=272342</a>
radar://124988039

Reviewed by Aditya Keerthi.

SystemPreviewController::begin was calling functions asychronously but
directly using both an Objective-C object and a C++ object without checking
to see if their lifetime&apos;s expired.

This code also used the name &apos;protectedThis&apos; to refer to a raw C++ pointer
which seems a little confusing, so just drop the protectedThis and use this
directly after ensuring the object has not been destructed based on the result
of WeakPtr::get().

* Source/WebKit/UIProcess/Cocoa/SystemPreviewControllerCocoa.mm:

Originally-landed-as: 272448.892@safari-7618-branch (b53cc7cedea8). <a href="https://rdar.apple.com/128279820">rdar://128279820</a>
Canonical link: <a href="https://commits.webkit.org/278999@main">https://commits.webkit.org/278999@main</a>
</pre>
----------------------------------------------------------------------
#### 6f6eb17ce35389deafb0be3cedfa5f9d0308ee94
<pre>
Interactionless USDZ Popups in Vision Pro
&lt;radar://122802255&gt;

Reviewed by Tim Horton.

Show a pop-up asking the user to confirm they want to
quick look preview a USDZ file in WebKit.

Add a delay if the page constantly tries to open one such
file otherwise it can be difficult to exit out of the page.

* Source/WebCore/en.lproj/Localizable.strings:
* Source/WebKit/UIProcess/Cocoa/SystemPreviewControllerCocoa.mm:
* Source/WebKit/UIProcess/SystemPreviewController.h:
* Source/WebCore/html/HTMLAnchorElement.cpp:
(WebCore::HTMLAnchorElement::handleClick):

Originally-landed-as: 272448.672@safari-7618-branch (6b6d3d3e61f6). <a href="https://rdar.apple.com/128090184">rdar://128090184</a>
Canonical link: <a href="https://commits.webkit.org/278998@main">https://commits.webkit.org/278998@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/91dae530544447517423b55b456f788399a9ed2b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/52196 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/31527 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/4616 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/55470 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/2919 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/54500 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/37927 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/2617 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/42481 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/1871 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/54292 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/29160 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/45049 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/23551 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/26428 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/2323 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/1078 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/48333 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/2469 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/57066 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/27321 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/2524 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/49873 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/28555 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/45165 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/49111 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11415 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/29467 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/28299 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski commented on Source/WebCore/html/HTMLAnchorElement.cpp:
```diff
@@ -608,7 +608,7 @@ void HTMLAnchorElement::handleClick(Event& event)
     SystemPreviewInfo systemPreviewInfo;
 #if USE(SYSTEM_PREVIEW)
-    systemPreviewInfo.isPreview = isSystemPreviewLink() && document->settings().systemPreviewEnabled();
+    systemPreviewInfo.isPreview = isSystemPreviewLink() && document->settings().systemPreviewEnabled() && UserGestureIndicator::processingUserGesture();
```
we want to revert this line

mwyrzykowski committed:
Interactionless USDZ Popups in Vision Pro
<radar://122802255>

Reviewed by Tim Horton.

Show a pop-up asking the user to confirm they want to
quick look preview a USDZ file in WebKit.

Add a delay if the page constantly tries to open one such
file otherwise it can be difficult to exit out of the page.

* Source/WebCore/en.lproj/Localizable.strings:
* Source/WebKit/UIProcess/Cocoa/SystemPreviewControllerCocoa.mm:
* Source/WebKit/UIProcess/SystemPreviewController.h:
* Source/WebCore/html/HTMLAnchorElement.cpp:
(WebCore::HTMLAnchorElement::handleClick):

Originally-landed-as: 272448.672@safari-7618-branch (6b6d3d3e61f6). rdar://128090184
Canonical link: https://commits.webkit.org/278998@main

mwyrzykowski committed:
Remove processing user gesture check before displaying prompt for usdz
https://bugs.webkit.org/show_bug.cgi?id=272321
radar://126078233

Reviewed by Alexey Proskuryakov.

Opening a usdz blob isn't considered to be processing a user gesture,
so having the UserGestureIndicator::processingUserGesture() check results
in the download path being taken instead of the prompt and open to ARQL.

Since we already have the prompt, we can remove the user gesture check.

* Source/WebCore/html/HTMLAnchorElement.cpp:
(WebCore::HTMLAnchorElement::handleClick):

Originally-landed-as: 272448.918@safari-7618-branch (fe35964ae2b9). rdar://128280210
Canonical link: https://commits.webkit.org/279000@main

mwyrzykowski committed:
Potential UAF in SystemPreviewController::begin
https://bugs.webkit.org/show_bug.cgi?id=272342
radar://124988039

Reviewed by Aditya Keerthi.

SystemPreviewController::begin was calling functions asychronously but
directly using both an Objective-C object and a C++ object without checking
to see if their lifetime's expired.

This code also used the name 'protectedThis' to refer to a raw C++ pointer
which seems a little confusing, so just drop the protectedThis and use this
directly after ensuring the object has not been destructed based on the result
of WeakPtr::get().

* Source/WebKit/UIProcess/Cocoa/SystemPreviewControllerCocoa.mm:

Originally-landed-as: 272448.892@safari-7618-branch (b53cc7cedea8). rdar://128279820
Canonical link: https://commits.webkit.org/278999@main
